src/org/ods/orchestration/TestStage.groovy

```diff
@@ -85,34 +67,32 @@ class TestStage extends Stage {
         }
         executeInParallel(executeRepos, generateDocuments)
 
-        // Update Jira issues with global data
-        jira.reportTestResultsForProject(
-            [Project.TestType.INSTALLATION],
-            globalData.tests.installation.testResults
-        )
-
-        jira.reportTestResultsForProject(
-            [Project.TestType.INTEGRATION],
-            globalData.tests.integration.testResults
-        )
-
-        jira.reportTestResultsForProject(
-            [Project.TestType.ACCEPTANCE],
-            globalData.tests.acceptance.testResults
-        )
-
-        // Parse all test report files into a single data structure
-        globalData.tests.acceptance.testResults = junit.parseTestReportFiles(
-            globalData.tests.acceptance.testReportFiles
-        )
-        globalData.tests.installation.testResults = junit.parseTestReportFiles(
-            globalData.tests.installation.testReportFiles
-        )
-        globalData.tests.integration.testResults = junit.parseTestReportFiles(
-            globalData.tests.integration.testReportFiles
-        )
+        globalData.tests.each {
```

Comment:
was this tested with marketplace etc ... this looks like making the $Itr error coming up again .. @s2oBCN @metmajer

Reply:
No, these changes were tested in a Box

Reply:
@s2oBCN please make sure the PR does not accidentally revert any NonSerializableException-related fixes. Once you checked, let's remove the Draft mode again. This is just to avoid last-minute errors creeping in again. Thanks!

Reply:
@s2oBCN - we fixed a bunch of $Itr issues in the teststage - and all this code is not in here right now ... ´largely via: 
https://github.com/opendevstack/ods-jenkins-shared-library/pull/770/files 

All that needs to go into master as well and merged with this one here

Reply:
@clemensutschig @metmajer 
I tested this with E2E tests, code has no serialization error.
For this kind of iteration there is Jenkins support https://issues.jenkins.io/browse/JENKINS-50997
Only what is important is to know that iterator is **Immutable**